src/io/platform/f3xx/f302x8/ADCf302x8.cpp

```diff
@@ -0,0 +1,252 @@
+#include <EVT/io/platform/f3xx/f302x8/ADCf302x8.hpp>
+#include <EVT/io/pin.hpp>
+
+#include <HALf3/stm32f3xx.h>
+
+#include <HALf3/stm32f3xx_hal_adc.h>
+#include <HALf3/stm32f3xx_hal_adc_ex.h>
+
+
+namespace {
+    /// This is made as a global variable so that it is accessible in the
+    // interrupt.
+    DMA_HandleTypeDef* dmaHandle;
+    ADC_HandleTypeDef* adcHandle;
+
+}  // namespace
+
+extern "C" void DMA1_Channel1_IRQHandler(void) {
+    HAL_DMA_IRQHandler(dmaHandle);
+}
+
+/**
+ * Current "work around" solution that ensurs the clock settings are applied
+ * and the HAL is initialized. This will later be moved into a common place
+ * that will happen once at system start up. For the time being this can be
+ * used for basic setup. This is not a long term solution.
```

Comment:
This better not be a long term solution :joy:

Reply:
Oh god no. After I2C im going to determine the best way for a system init. This is trash right now

Reply:
If the ```feature/system_clock_init``` branch still exists in the old repo, feel free to copy-paste those clock settings for each target. If the designs of the boards haven't changed dramatically (don't see why they'd need to) then those values should still be valid. Otherwise you can generate them yourself with STM32CubeMX.